PYTHON/activityLog/activityLog.py

```diff
@@ -0,0 +1,356 @@
+import os
+import sys
+import shutil
+import subprocess
+import platform
+import re
+import argparse
+from datetime import datetime
+import time
+import tarfile
+
+### TODO: redesign as a class object (OOP)
+
+class ActivityLogger(object):
+	"""Generates a log file for server activity, CPU load, and MySQL statistics."""
+	def __init__(self):
+		super(ActivityLogger, self).__init__()
+		self.checkcronjob()
+		self.epochflag = False
+		self.rootcheck = False
+		self.mysql_usr = None
+		self.mysql_pass = None
+		self.mysql_queries = 0
+		self.mysql_interval = 0
+		self.logfilename = None
+		### Get server specs
+		self.current_platform = platform.platform().lower() # platform information in lower case
+
+		
+	def checkcronjob(self):
+		# Checks to see if script was run by user, or a cron job.
+		# Will create actLogDir in home dir if cron, current working directory 
+		# if run by user.
+		# Explained: sys.stdin will be a TTY if run by console (by the user).
+		if os.isatty(sys.stdin.fileno()):
+			self.cronjob = False
+			self.silent_mode = False
+		else:
+			self.cronjob = True
+			self.silent_mode = True
+
+
+	# Returns a tuple of the stdout and stderr of a command. 
+	# NOTE that the use of shell=True is extremely dangerous from a security standpoint. Using it
+	# here because the alternative (parsing each command and argument into a list) makes piping
+	# multiple commands difficult.
+	def run_command(self, cmd):
+		return subprocess.Popen(
+			cmd, 
+			shell=True, 
+			stdout = subprocess.PIPE, # stdout in [0]
+			stderr = subprocess.PIPE).communicate() # stderr in [1]
+
+	def dateString(self, epoch = False):
+		# returns time from epoch.  ON UNIX, it is 0 hours on Jan. 1st, 1970
+		if epoch == True: 
+			return time.time()
+		elif epoch != True:
+		# using UTC (Similar to GMC)
+		# strftime converts the struct_time format into a string. see documentation
+			dateTime = datetime.utcnow().strftime("%mm_%dd_%Yy__at_UTC_%Hh_%Mm_%Ss")
```

Comment:
Timezones will bite you in the ass many times in your career, I guarantee it.  You are definitely on the right track here using `utcnow()`, good job!  But, as you probably noticed, it returns a timestamp with no timezone (though the time is correct for UTC)!  So I always make sure to add the tzinfo from the essential `pytz` package: 
```
   right_the_fuck_now = datetime.utcnow().replace(tzinfo=pytz.UTC)
```

Also, for formatting, there's a shorthand method `isoformat()` that you should get under your fingers too.  
```
>>> datetime.utcnow().replace(tzinfo=pytz.UTC).isoformat()
'2015-12-17T20:38:09.327356+00:00'
```